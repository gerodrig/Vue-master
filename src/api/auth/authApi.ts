import axios from "axios";


export default axios.create({
    baseURL: 'https://identitytoolkit.googleapis.com/v1/accounts',
    params: {
        key: import.meta.env.VITE_FIREBASE_API_KEY,
    },
});

