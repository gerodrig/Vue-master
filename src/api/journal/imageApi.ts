import { v2 as cloudinary } from 'cloudinary';

cloudinary.config({
    cloud_name: import.meta.env.VITE_CLOUDINARY_CLOUD_NAME,
    api_key: import.meta.env.VITE_CLOUDINARY_API_KEY,
    api_secret: import.meta.env.VITE_CLOUDINARY_API_SECRET,
    secure: true,
  });

export const deleteImage = async (url: string | null): Promise<boolean> => {

    if(!url) return false;

    if (url) {
        //url format https://res.cloudinary.com/gerodrig18/image/upload/v1682391219/vue-test/wbk4pifuthuyxqn08ly1.jpg

        // get vue-test/wbk4pifuthuyxqn08ly1
        // const public_id = url.split('/').pop()?.split('.')[0] || '';
        const public_id = url.split('/').slice(-2).join('/').split('.')[0] || '';
  
        try {
          const result = await cloudinary.api.delete_resources([public_id], {});
            //check if deleted: {'deleted': {'vue-test/': 'deleted'}} includes deleted return true
            //else return false
            if(Object.values(result.deleted)[0] === 'deleted') return true;
            
        } catch (error) {
          console.log(error);
        }
      }

      return false;
};