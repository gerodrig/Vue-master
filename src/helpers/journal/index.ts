export {default as uploadImage} from './uploadImage';
export { getDate } from "./getDate";


