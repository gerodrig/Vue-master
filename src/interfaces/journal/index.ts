export interface JournalState {
        isLoading: boolean;
        entries: Entry[];
}

export interface AuthState {
    status: AuthStatus;
    user: null | User;
    idToken: null | string;
    refreshToken: null | string;
}

export interface User {
    username?: string;
    email: string;
    password?: string;
}


//enum for the auth status
export type AuthStatus = 'authenticated' | 'not-authenticated' | 'authenticating';

export interface Rootstate {
    journal?: JournalState;
    auth?: AuthState;
}

export interface Entry {
    id: string;
    date: string | number;
    text: string;
    picture?: string | null;
}

export interface FormattedDate {
    day: number;
    month: string;
    year: number;
    dayOfWeek: string;
  }

export interface EntryFirebase {
    [ key: string ]: {
        date: string;
        text: string;
        picture: string | null;
    }
}