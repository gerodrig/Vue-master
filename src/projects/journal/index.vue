<template>
  <Header :title="title" />
  <router-view />

</template>

<script lang="ts">
import useAuth from '@/composables/auth/useAuth';
import { computed, defineAsyncComponent, watch } from 'vue';
import { useRouter } from 'vue-router';

export default {
  name: 'Journal App',
  components: {
    Header: defineAsyncComponent(() => import('@/components/shared/Header.vue')),
  },
  setup() {
    const { checkAuthStatus, authStatus } = useAuth();
    const router = useRouter();

    watch(authStatus, () => {
      if(authStatus.value !== 'authenticated') {
        router.push({ name: 'login' });
      }
    });

    checkAuthStatus();
    return {
      authStatus,
      title: computed(() => 'Journal App')
    };
  },
};
</script>
