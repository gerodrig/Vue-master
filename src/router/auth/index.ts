export default {
    name: 'auth',
    component: () => import(/* webpackChunkName: "auth" */ '@/auth/journal/layouts/AuthLayout.vue'),
    children: [
        {
            path: 'login',
            name: 'login',
            component: () => import(/* webpackChunkName: "Login" */ '@/auth/journal/views/Login.vue'),
        },
        {
            path: 'register',
            name: 'register',
            component: () => import(/* webpackChunkName: "Resgister" */ '@/auth/journal/views/Register.vue'),
        },
    ],
}