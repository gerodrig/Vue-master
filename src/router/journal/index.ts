import { RouteLocationNormalized as Route } from 'vue-router';

export const RouteNames = {
    HOME: 'daybook',
    ENTRY: 'entry',
    NOENTRY: 'no-entry',
  }

export default {
    name: RouteNames.HOME,
    component: () => import(/* webpackChunkName: "daybook" */ '@layouts/journal/DayBookLayout.vue'), 
    children: [
        {
            path: '',
            name: RouteNames.NOENTRY,
            component: () => import(/* webpackChunkName: "daybook-no-entry" */ '@/views/journal/NoEntrySelected.vue') 

        },
        {
            path: ':id',
            name: RouteNames.ENTRY,
            component: () => import(/* webpackChunkName: "daybook-entry" */ '@/views/journal/EntryView.vue'),
            props: (route: Route) => {
                return {
                    id: route.params.id
                }
            }
        }
    ],
}

