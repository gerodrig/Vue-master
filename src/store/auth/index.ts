
import state from './state';
import * as getters from './getters';
import * as mutations from './mutations';
import * as actions from './actions';


const authModule = {
    namespaced: true,
    actions,
    getters,
    mutations,
    state,
};

export default authModule;