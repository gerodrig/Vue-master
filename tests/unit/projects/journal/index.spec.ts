import { describe, expect, it, vi, beforeEach } from 'vitest';
import { shallowMount, VueWrapper } from '@vue/test-utils';
import { createStore } from 'vuex';
import { useRouter } from 'vue-router';

import journalStore from '@/store/journal';
import authStore from '@/store/auth';
import journal from 'tests/unit/mocks/journal/test-journal-state';
import auth from 'tests/unit/mocks/auth/test-auth-state';

import Home from '@/projects/journal/index.vue';

import { Rootstate } from '@/interfaces/journal/index';

//Mocking router
vi.mock('vue-router', () => ({
  useRouter: vi.fn(() => ({
      push: vi.fn()
  }))
}));

//Mocking store
const createVuexStore = (initialState: Rootstate) =>
  createStore({
    modules: {
      journal: {
        ...journalStore,
        state: { ...initialState },
      },
      auth: {
        ...authStore,
        state: { ...initialState },
      },
    },
  });

describe('Tests in Main page', () => {
  const store = createVuexStore({
    journal,
    auth,
  });

  let wrapper: VueWrapper;

  beforeEach(() => {
    vi.clearAllMocks();
    wrapper = shallowMount(Home, {
      global: {
        stubs: ['router-link', 'router-view'],
        plugins: [store],
      },
    });    
  });

  it('must match with snapshot', () => {   

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should test that routerview content is rendered', async () => {

    const routerView = wrapper.find('router-view-stub');

    expect(routerView.exists()).toBeTruthy();
  });
});
