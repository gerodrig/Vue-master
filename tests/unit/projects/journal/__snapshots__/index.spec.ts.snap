// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in Main page > must match with snapshot 1`] = `
"<header-stub title=\\"Journal App\\"></header-stub>
<router-view-stub></router-view-stub>"
`;
